src/app/container/email.service.ts

```diff
@@ -11,47 +11,15 @@ export class EmailService {
    * @param subject The subject of the email
    * @param body The body of the the email
    */
-  private composeEmail(email: string, subject: string, body: string): string {
+  private static composeEmail(email: string, subject: string, body: string): string {
     return `mailto:${email}?subject=${subject}&body=${body}`;
   }
 
-  /**
-   * Composes the href for the Request Access email
-   * @param tool The tool to request access for
-   */
-  public composeRequestAccessEmail(tool: ExtendedDockstoreTool): string {
-    const email = this.getRequestEmailMailTo(tool.tool_maintainer_email, tool.email);
-    const subject = this.getRequestEmailSubject(tool.tool_path);
-    const body = this.getRequestEmailBody(tool.tool_path, tool.imgProvider);
-    return this.composeEmail(email, subject, body);
-  }
-
-  /**
-   * Composes the href for the Contact Author email
-   * @param tool The tool to contact author for
-   */
-  public composeContactAuthorEmail(tool: ExtendedDockstoreTool): string {
-    const email = this.getInquiryEmailMailTo(tool.email);
-    const subject = this.getInquiryEmailSubject(tool.tool_path);
-    const body = this.getInquiryEmailBody();
-    return this.composeEmail(email, subject, body);
-  }
-
-  // Request Access button methods
-  /**
-   * This gets the mailto address when the Request Access button is clicked
-   * @param maintainerEmail The maintainer email
-   * @param email The tool author's email address
-   */
-  private getRequestEmailMailTo(maintainerEmail: string, email: string): string {
-    return this.dockstoreService.getRequestAccessEmail(maintainerEmail, email);
-  }
-
   /**
    * This gets the email subject when the Request Access button is clicked
    * @param path The path of the tool whose access is being requested
    */
-  private getRequestEmailSubject(path: string): string {
+  private static getRequestEmailSubject(path: string): string {
```

Comment:
Why did you make this and other methods static?

Reply:
Gary suggested to make some functions in this file static so testing is easier. I think this is what he is talking about

Reply:
@coverbeck To be honest I'm not sure myself, but my understanding is that since these functions are static, we don't need to instantiate them over and over again along side with the instantiation of the class object. Plus, since all of these class objects have access to the functions, and they should store the same info across, so there's no need to pass in new parameters. The above is my understanding, it probably doesn't have anything to do with testing in specific. But I guess Gary could explain better? @garyluu

Reply:
@coverbeck Email.service.ts is essential a file full of helper functions. The functions should not have any side effects (like subscriptions).  It's made to static in the event others are tempted to do something like subscribe to tokens/users within the file. Can you explain why you don't want it static?

Reply:
I'm wasn't necessarily against it. I just saw the changes to add static, but didn't see any code that relied on the change (or am I wrong?), and was wondering why it was changed.

Your explanation " It's made to static in the event others are tempted..." works for me.

Reply:
@garyluu The thing is I did not make all private functions static. There's one that needs to access this.something, so I left it as it is

Reply:
@wendidy yes, i'm well aware not all of them were changed (even though the could've). not really a high priority, so I didn't request changes for that